### PR TITLE
Update all submodules

### PR DESCRIPTION
**Pull request by joshuahansel**
Refs #0

### CODE REVIEW

**Comment by joshuahansel** [comment]
A MOOSE patch is needed - I currently added a version from my fork with some fixes (not merged yet).

**Comment by GiudGiud** [comment]
looks like there are some 'speed' functor failures. So this requires your patch to moose to make it to pgh and then a pgh subdmodule update

**Comment by joshuahansel** [comment]
Right, I only did that for BC and was basically not looking at PH results, since that should just be a subset-ish. I still need to fix some BC tests it looks like. It was just difficult to get everything fixed without having SAM to compile into BC.

Could you update SAM for me, since I don't have access? We'd need to wait for SAM to get the MOOSE update too.

I should be able to do Cardinal, since it's open.

**Comment by joshuahansel** [comment]
For the Lotus test that is timing out: I saw that in `tests` a file called `run_neutronics.i` was used that doesn't exist - appears to have been replaced by `run_neutronics_9_group.i`, so I changed this.

I'll see if I can use the BC on HPC to regold the diffing test, since that one has SAM.

**Comment by joshuahansel** [comment]
That lotus test is still timing out, even after doubling the max time to 2000 s. It already has a minimum of 32 processes, and it's a steady test, so I can't just reduce the number of time steps, so I'm not sure what we should do to resolve it.

I'll address the remaining Direwolf failure.

**Comment by joshuahansel** [comment]
Direwolf failure is because DW needs a Sockeye update, so I'll wait on https://github.inl.gov/ncrc/dire_wolf/pull/294 and then update DW again.

**Comment by joshuahansel** [comment]
There is a test, `sfr/abtr_xsgen_workflow.xs_generation_syntax` (https://civet.inl.gov/job/2294439/) that uses `Mesh/data_driven_generator`, which apparently requires the app to set `allow_data_driven_mesh_generation` to true because it's experimental. What's the move here? Change the test or set the `allow_data_driven_mesh_generation` to `true` in Griffin/BlueCrab? @loganharbour @shikhar413

**Comment by GiudGiud** [comment]
Should already be set in Griffin.
I guess we need to set it in BlueCrab too.

**Comment by shikhar413** [comment]
If there is a BlueCrabApp.C, you can add the following line in validParams:

```
// Needed for enabling data driven mesh generation
params.set<bool>(MeshGeneratorSystem::allow_data_driven_param) = true;
```

**Comment by joshuahansel** [comment]
I will add it to BlueCrabApp.

**Comment by joshuahansel** [comment]
@GiudGiud would you mind reviewing https://github.inl.gov/ncrc/blue_crab/pull/281?

**Comment by joshuahansel** [comment]
Hopefully https://github.com/idaholab/civet_recipes/pull/1751 should address the BC failure.

File: htgr/generic-pbr-tutorial/gold/step7_out.csv
Comment (GiudGiud) [inline htgr/generic-pbr-tutorial/gold/step7_out.csv:1]: For traceability purposes, these regolds are necessary due to the change in the friction factor definition.
Given that the values are arbitrary in the tutorial for example:
```
  [drag_cavity]
    type = ADGenericVectorFunctorMaterial
    prop_names = 'Darcy_coefficient Forchheimer_coefficient'
    prop_values = 'c_drag c_drag c_drag 0 0 0'
    block = 'cavity'
  []
```
updating to adapt the new definition does not make much sense

**Comment by GiudGiud** [comment]
https://github.com/idaholab/moose/pull/27951 has a fix that will likely prevent regolding these MSFR tests. I'll get it green again

**Comment by GiudGiud** [comment]
ok new failures are fixed by this PR
https://github.com/idaholab/moose/pull/28096

there's no "previous submodule" we can use because the "removing NSFV action" both creates these new issues and removes the need for the regold of the MSFR input

**Comment by GiudGiud** [comment]
Needs a submodule update still, Pronghorn DW and BlueCrab dont have an up-to-date enough moose

**Comment by GiudGiud** [comment]
Only failure is an intermittent timeout

**Review by GiudGiud — APPROVED** [review]

**Pull request merged**